### ISSUE TITLE
[ENH] interface `TweedieRegressor` from `sklearn` as `skpro` regressor

### ISSUE THREAD

**Issue by fkiraly** [sktime/skpro#423]
We should try to interface `TweedieRegressor` from `sklearn` as an `skpro` regressor.
https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.TweedieRegressor.html

Notes on implementation:

* the current adapter will not work because it does not follow the `return_std` interface, but we can use `_prep_skl_df`.
* We would need a `Tweedie` distribution in `skpro`, currently it is not implemented.
* `Tweedie` has three parameters: power, location, scale. Power is set fixed in the `sklearn` `TweedieRegressor`, location is returned by `predict`, but it is unclear whether scale can be obtained from it. Perhaps @fsaforo1 has insight on this point.

FYI @ShreeshaM07, this is very similar to your previous work on `statsmodels` GLM!

**Comment by ShreeshaM07** [comment]
> this scipy issue discusses the Tweedie distribution: https://github.com/scipy/scipy/issues/11291#issuecomment-1868256070 and concludes that the scipy interface is not general enough because it is mixed type. skpro is general enough, so with the pointers in there we could implement it, either entirely from scratch, or interfacing some of the component functions such as Bessel.

From the conversation I can infer that we can implement this in `skpro` as it allows for mixed type distributions with `pdf` and `pmf` in different intervals. https://lorentzen.ch/index.php/2024/06/17/a-tweedie-trilogy-part-iii-from-wrights-generalized-bessel-function-to-tweedies-compound-poisson-distribution/ seems to be a very informative post explaining the Tweedie distribution. It also gives code snippet for the pdf and pmf of the function compound poisson and gamma function.
```
import numpy as np
from scipy.special import wright_bessel


def cpg_pmf(mu, phi, p):
    """Compound Poisson Gamma point mass at zero."""
    return np.exp(-np.power(mu, 2 - p) / (phi * (2 - p)))

def cpg_pdf(x, mu, phi, p):
    """Compound Poisson Gamma pdf."""
    if not (1 < p < 2):
        raise ValueError("1 < p < 2 required")
    theta = np.power(mu, 1 - p) / (1 - p)
    kappa = np.power(mu, 2 - p) / (2 - p)
    alpha = (2 - p) / (1 - p)
    t = ((p - 1) * phi / x)**alpha
    t /= (2 - p) * phi
    a = 1 / x * wright_bessel(-alpha, 0, t)
    return a * np.exp((x * theta - kappa) / phi)
```
This can be utilized along with the usage of the `wright_bessel` function in `scipy.special`.

> for the sklearn Tweedie regressor, the remaining quesiton is still where to get the scale from. It would not be much of a Tweedie regressor if tha twould be impossible to obtain...

Can this not be obtained using the below ?
![image](https://github.com/user-attachments/assets/7472c06e-8a0e-4be9-bd75-feb905c918c0)
As we know the mean=return of `predict` we know `p` power parameter is fixed. We can calculate `phi` using the formula below  thereby finding out the `std deviation` . Is it not possible that way?


​

**Comment by ShreeshaM07** [comment]
Some thought on the Tweedie Distribution
- Since it is distinguished into type of distribution using the power parameter itself we can just call pdf of `Normal` when `pw=0` where `pw` is the power parameter, call pdf of `Poisson` when `pw=1`, pdf of Gamma when `pw=2` and call the code snippet in the above <a href=https://github.com/sktime/skpro/issues/423#issuecomment-2231637334>comment</a> when `p is in (1,2)`
![image](https://github.com/user-attachments/assets/e9a18c31-bc04-454b-9455-b164c3712e15)

**Comment by fkiraly** [comment]
> From the conversation I can infer that we can implement this in skpro as it allows for mixed type distributions with `pdf` and `pmf` in different intervals.

Yes, assuming you mean the `p` parameter. In places where the distribution is entirely discrete or continuous, the `pdf` or `pmf` will return zero.

Further, here's an interesting option, since multiple already implemented distributions figure as special cases:

* we could implement the individual families separately, e.g., compound Poisson-Gamma
* define `Tweedie` as a `_DelegatedDistribution` and delegate to one of the families depending on `p`.
* as you say, we need to ensure that the parameters are mapped correctly, e.g., Tweedie being parameterized by mu, sigma, and Gamma by alpha, beta.
* probably we also want to change the `_DelegatedDistribution` to delegate private, not public methods. This could be done in a separate PR - the current delegator

Here is an illustration of the suggested delegator approach:
![image](https://github.com/user-attachments/assets/ccd482b1-9d3e-4cf0-87a2-3fa0adbd964c)

**Comment by fkiraly** [comment]
Opened new issue on Tweedie distribution here, as that does not seem too straightforward - for further discussion.
